libs/public-pages/src/pages/file-publication/index.ts

```diff
@@ -0,0 +1,46 @@
+import { getArtefactById, getUploadedFile, mockListTypes } from "@hmcts/publication";
+import { formatDateAndLocale } from "@hmcts/web-core";
+import type { Request, Response } from "express";
+import { cy } from "../artefact-not-found/cy.js";
+import { en } from "../artefact-not-found/en.js";
+
+export const GET = async (req: Request, res: Response) => {
+  const artefactId = req.query.artefactId as string;
+  const locale = res.locals.locale || "en";
+  const t = locale === "cy" ? cy : en;
+
+  console.log("[file-publication] Received request for artefactId:", artefactId);
+
+  if (!artefactId) {
+    console.log("[file-publication] Missing artefactId, redirecting to 400");
+    return res.redirect("/400");
+  }
+
+  const file = await getUploadedFile(artefactId);
+
+  if (!file) {
+    console.log("[file-publication] File not found for artefactId:", artefactId, "rendering error page");
+    return res.status(404).render("artefact-not-found/index", t);
```

Comment:
I t might be cleaner if we redirect to 'artefact-not-found' page rather than rendering the page directly in this controller. There is another occurrence below.